### DISCUSSION TITLE
openAPI spec generation - defining responses on an operation

### DISCUSSION THREAD

**Discussion by MCR2019** [aws-powertools/powertools-lambda-python#3540] [General]
Really excited about the quite new powertools openAPI spec generation so thank to all those who've worked on it and those working on it now :) 

I'm trying to get it working for an endpoint and following the documentation on how to specify responses here:
[Customizing API Operations docs](https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/#customizing-api-operations)

Specifically I'd like to add custom reponses which the documentation shows as:

```
@app.get(
    "/todos/<todo_id>",
    summary="Retrieves a todo item",
    description="Loads a todo item identified by the `todo_id`",
    response_description="The todo object",
    responses={
        200: {"description": "Todo item found"},
        404: {
            "description": "Item not found",
        },
    },
    tags=["Todos"],
)
def get_todo_title(todo_id: int) -> str:
    todo = requests.get(f"https://jsonplaceholder.typicode.com/todos/{todo_id}")
    todo.raise_for_status()
```

I'd really like to be able to specify the content/schema of the response, maybe as a Pydantic Model which would then generate the response schema in the openAPI spec. (in the same way that the function response type is used).

At the moment I'm only able to get it to work with a basic string which doesn't represent the actual return type/structure.

Is there any more info on what the intention is for how this should be used with a fuller example?

I'm aware that this functionality is quite new so I'm not sure if this is something already on the to do list?

Any info would be great, and thanks again!

**Comment by rubenfonseca** [comment]
Hi @MCR2019 this makes sense, [FastAPI supports it this way](https://fastapi.tiangolo.com/advanced/additional-responses/?h=responses), so we could work on adding support in Powertools too. I'm converting this to a Feature Request ticket!

**Comment by rubenfonseca** [comment]
Closing in favour of https://github.com/aws-powertools/powertools-lambda-python/issues/3542
